bob-backend/src/pearl/disk_controller.rs

```diff
@@ -290,13 +290,15 @@ impl DiskController {
         Ok(groups)
     }
 
-    async fn find_group(&self, operation: &Operation) -> BackendResult<Group> {
-        Self::find_in_groups(self.groups.read().await.iter(), operation).ok_or_else(|| {
-            Error::failed(format!("cannot find actual alien folder. {:?}", operation))
-        })
+    async fn find_all_groups(&self, operation: &Operation) -> Vec<Group> {
```

Comment:
Most of the time the result will contain 0 or 1 record. I suggest to use `SmallVec<>` here to improve performance slightly